src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -163,12 +165,11 @@ private void addBinaryToPath(String hdfsBinaryPath) throws IOException {
    **/
   protected void startProcess(ExecutorDriver driver, Task task) {
     reloadConfig();
-    Process process = task.process;
-    if (process == null) {
+    if (task.process == null) {
       try {
-        process = Runtime.getRuntime().exec(new String[]{
-            "sh", "-c", task.cmd});
-        redirectProcess(process);
+        ProcessBuilder processBuilder = new ProcessBuilder("sh", "-c", task.cmd);
+        task.process = processBuilder.start();
+        redirectProcess(task.process);
       } catch (IOException e) {
         log.fatal(e);
         sendTaskFailed(driver, task);
```

Comment:
Good point, I am adding code to destroy the process and send task failed as opposed to destroying the executor completely.